qiskit/pulse/instructions/reference.py

```diff
@@ -86,6 +86,11 @@ def channels(self) -> tuple[Channel, ...]:
         """Returns the channels that this schedule uses."""
         raise UnassignedReferenceError(f"Subroutine is not assigned to {self.ref_keys}.")
 
+    @property
+    def inst_targets(self):
+        """Returns the objects targeted by the instruction."""
+        raise NotImplementedError
```

Comment:
This should be `UnassignedReferenceError`. Once reference is assigned to the schedule block this instruction is replaced with reference to the subroutine (another schedule block) that returns valid channels.